lib/openapi_client_generator.rb

```diff
@@ -0,0 +1,241 @@
+      if definition.request_body
+        params += definition.request_body.properties_for_format("application/json").reject { |param| definition.request_body.content["application/json"]["schema"]["required"].include? param.name }
+      end
+      params
+    end
+
+    def parameter_type(param)
+      {
+        "repo" => "[Integer, String, Repository, Hash]",
+      }[param.name] || "[#{param.type.capitalize}]"
+    end
+
+    def parameter_description(param)
+      return "A GitHub repository" if param.name == "repo"
+      return "The ID of the #{param.name.gsub("_id", "").gsub("_", " ")}" if param.name.end_with? "_id"
+      return param.description.gsub("\n", "")
+    end
+
+    def parameter_documentation
+      required_params.map {|param|
+        "@param #{param.name} #{parameter_type(param)} #{parameter_description(param)}"
+      } + optional_params.map {|param|
+        "@param options [#{param.type.capitalize}] :#{param.name} #{param.description.gsub("\n", "")}"
+      }
+    end
+
+    def return_value_description
+      case verb
+      when "GET"
+        if singular?
+          "A single #{namespace.gsub("_", " ")}"
+        else
+          "A list of #{namespace.gsub("_", " ")}"
+        end
+      when "POST"
+        "The new #{namespace.singularize.gsub("_", " ")}"
+      else
+      end
+    end
+
+    def verb
+      definition.method.upcase
+    end
+
+    def parameters
+      parameterizer.parameterize(required_params.map(&:name))
+    end
+
+    def namespace
+      definition.operation_id.split("/").last.split("-").drop(1).join("_")
+    end
+
+    def method_name
+      case verb
+      when "GET"
+        namespace
+      when "POST"
+        "create_#{namespace}"
+      else
+      end
+    end
+
+    def alternate_name
+      return unless verb == "GET"
+      return if singular?
+      "list_#{namespace}"
+    end
+
+    def parts
+      definition.path.path.split("/").reject { |segment| segment.include? "id" }
+    end
+
+    def priority
+      [parts.count, VERB_PRIORITY.index(verb), singular?? 0 : 1]
+    end
+  end
+
+  class API
+    def self.at(definition, parameterizer: OpenAPIClientGenerator::Endpoint::PositionalParameterizer)
+      grouped_paths = definition.paths.group_by do |oas_path|
+        resource_for_path(oas_path.path)
+      end
+      grouped_paths.delete(:unsupported)
+      grouped_paths.each do |resource, paths|
+        endpoints = paths.each_with_object([]) do |path, arr|
+          path.endpoints.each do |endpoint|
+            arr << OpenAPIClientGenerator::Endpoint.new(endpoint, parameterizer: parameterizer)
+          end
+        end
+        yield new(resource, endpoints: endpoints)
+      end
+    end
+
+    def self.resource_for_path(path)
+      return :unsupported unless path.include? "deployment"
+      path_segments = path.split("/").reject{ |segment| segment == "" }
+      resource = path_segments[3]
```

Comment:
This seems pretty magic, does this ever change?

Reply:
It definitely will, but it's pretty consistent for the paths under "repos" which was my next focus. I can look further into it, but I felt like it was good enough for now 🤔